api-guidelines/v4.0/API-DESIGN-GUIDELINES/DISCOVERY-API.md

```diff
@@ -0,0 +1,170 @@
+# Discovery API
+
+The default response when retrieving the base endpoint on an Ed-Fi
+implementation must be a JSON document that provides information about the
+application version, supported data model(s), and URLs for additional metadata.
+Historically, this root-level metadata resource was known as the "version
+endpoint".
+
+> [!TIP]
+> The Discovery API is completely described in an
+> [Open API 3 specification document](../../../api-specifications/discovery-api/).
+
+The following sample JSON demonstrates a typical example of the document:
+
+```json
+{
+    "version": "7.1",
+    "informationalVersion": "7.1",
+    "suite": "3",
+    "build": "7.1.1294.0",
+    "dataModels": [
+        {
+            "name": "Ed-Fi",
+            "version": "5.0.0",
+            "informationalVersion": "The Ed-Fi Data Model 5.0"
+        },
+        {
+            "name": "TPDM",
+            "version": "1.1.0",
+            "informationalVersion": "TPDM-Core"
+        }
+    ],
+    "urls": {
+        "dependencies": "https://api.ed-fi.org/v7.1/api/metadata/data/v3/dependencies",
+        "openApiMetadata": "https://api.ed-fi.org/v7.1/api/metadata/",
+        "oauth": "https://api.ed-fi.org/v7.1/api/oauth/token",
+        "dataManagementApi": "https://api.ed-fi.org/v7.1/api/data/v3/"
+    }
+}
+```
+
+The example above is based on [version
+1.0](../../../api-specifications/discovery/1.0/) specification, used by the
+Ed-Fi ODS/API Platform. New Ed-Fi API implementations are encouraged to look
+into the modifications in the [draft version
+2.0](../../../api-specifications/discovery/2.0-draft/).
+
+## Root-Level Information
+
+The root-level information (in this example, `version`, `informationalVersion`,
+`suite`, and `build`) describes the deployed application. If the application
+complies with these design guidelines and the relevant API specification, then
+most _client applications_ should not need to know which _server_ application it
+is communicating with. However, this information can be valuable when debugging
+the client-server interaction.
+
+## Data Models
+
+This section lists all of the Data Models that are served by this API
+application. This will always include the core Ed-Fi (Unifying) Data Model.
+Other models listed are the _extensions_ that a given deployment implements. The
+example above shows the core Teacher Preparation Data Model (TPDM) extension.
```

Comment:
might be worth defining/reserving `ed-fi`, maybe requiring it since I don't think someone is really an Ed-Fi API if they don't implement the Ed-Fi UDM. 

I also might explicitly tie the value of the data model ("ed-fi", "tpdm", "TX") to the "namespace" (schema?) used in URLs (e.g. the tpdm in `/tpdm/candidateEducatorPreparationProgramAssociations`) or extensions (e.g. `_ext: [ "tpdm": ... ]`)

Reply:
I am going to change to "must" instead of "will always" on the UDM. However, a future revision may alter this a bit. There is a notion out there of "subsets" of the UDM, such as the Finance API or the Assessment API. In theory someone should be able to implement only that subset. Maybe an Assessment provider creates a pull API, and they don't bother implementing the full UDM. It might be useful to indicate that here, but that's a challenge for another day.